### PR TITLE
feat(FN-3579): add record correction data change links

### PR DESCRIPTION
**Pull request by sw-mattw**
## Introduction :pencil2:
The "check the information" screen in the record correction journey requires "change" links be added to the rows related to fields the user has entered data for. This should return to the previous screen, with the form values persisted.

## Resolution :heavy_check_mark:
- Added "change" links to the two form field rows
- Added component tests to check link href and text
- Added e2e test to ensure that clicking the change button returns to the previous screen with the entered form values persisted

"Check the information" page - note the two "Change" links to the right of the 'reason' and 'more information' rows:
FN-3579_PR-1_check-the-information-page-links

Clicking on these "Change" links returns to the prior page with form values persisted (note that the persistence work was covered by #3961 and #4022):
FN-3579_PR-1_change-link-destination

## Miscellaneous :heavy_plus_sign:
N/A

### CODE REVIEW

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js [hunk 1]
```diff
@@ -220,5 +220,39 @@ context('When fee record correction feature flag is enabled', () => {
...
+        cy.url().should('eq', relative(`/utilisation-reports/${reportId}/create-record-correction-request/${feeRecordAtToDoStatus.id}/check-the-information`));
+
+        checkFeeRecordCorrectionRequestPage.clickReasonsChangeLink();
+      });
+
+      it('should return to the "create record correction request" screen with the form populated with the prior form data', () => {
+        cy.url().should('eq', relative(`/utilisation-reports/${reportId}/create-record-correction-request/${feeRecordAtToDoStatus.id}`));
+
+        createFeeRecordCorrectionRequestPage.reasonCheckbox(RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT).should('be.checked');
+        createFeeRecordCorrectionRequestPage.reasonCheckbox(RECORD_CORRECTION_REASON.OTHER).should('be.checked');
```
Comment (Zainzzkk) [inline e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js:249]: could you add an empty line here, just to split between checked and not checked please

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js [hunk 2]
```diff
@@ -220,5 +220,39 @@ context('When fee record correction feature flag is enabled', () => {
...
+        createFeeRecordCorrectionRequestPage.reasonCheckbox(RECORD_CORRECTION_REASON.REPORTED_FEE_INCORRECT).should('not.be.checked');
+        createFeeRecordCorrectionRequestPage.reasonCheckbox(RECORD_CORRECTION_REASON.REPORTED_CURRENCY_INCORRECT).should('not.be.checked');
+        createFeeRecordCorrectionRequestPage.reasonCheckbox(RECORD_CORRECTION_REASON.UTILISATION_INCORRECT).should('not.be.checked');
+
+        cy.assertText(createFeeRecordCorrectionRequestPage.additionalInfoInput(), 'Some additional info.');
```
Comment (Zainzzkk) [inline e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js:254]: could 'Some additional info.' be a const, same as line 236 so we only change it in one place?

File: trade-finance-manager-ui/component-tests/utilisation-reports/record-corrections/check-the-information.component-test.ts [hunk 3]
```diff
@@ -88,7 +88,25 @@ describe('page', () => {
...
-  it('should render the provided more information', () => {
+  it('should render the "reason for record correction" change link', () => {
+    // Arrange
+    const reportId = '123';
+    const feeRecordId = '456';
+    const viewModel: RecordCorrectionRequestInformationViewModel = {
+      ...aRecordCorrectionRequestInformationViewModel(),
+      reportId,
+      feeRecordId,
+    };
```
Comment (Zainzzkk) [inline trade-finance-manager-ui/component-tests/utilisation-reports/record-corrections/check-the-information.component-test.ts:99]: can these be at the top? they are repeated

**Review by RLavender98 — DISMISSED** [review]
LGTM 😄

File: trade-finance-manager-ui/component-tests/utilisation-reports/record-corrections/check-the-information.component-test.ts (hunk 3 above)
Comment (sw-mattw) [reply trade-finance-manager-ui/component-tests/utilisation-reports/record-corrections/check-the-information.component-test.ts]: Pulled `reportId` and `feeRecordId` up thanks, kept the view model setup here as this is specific for each test and uses the `...aRec`... test helper to provide a base view model to build on

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js (hunk 2 above)
Comment (sw-mattw) [reply e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js]: Ah good point thanks, pulled this to the top of this context block and replaced all literal references

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js (hunk 1 above)
Comment (sw-mattw) [reply e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js:282]: Good point, done this for the other cases as well thanks

File: e2e-tests/tfm/cypress/e2e/pages/utilisation-reports/fee-record-correction/checkFeeRecordCorrectionRequestPage.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const checkFeeRecordCorrectionRequestPage = {
+  clickReasonsChangeLink: () => cy.get(`a[data-cy="change-record-correction-reason-link"]`).click(),
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply e2e-tests/tfm/cypress/e2e/pages/utilisation-reports/fee-record-correction/checkFeeRecordCorrectionRequestPage.js]: I don't see this change yet @sw-mattw

**Review by Zainzzkk — APPROVED** [review]

File: e2e-tests/tfm/cypress/e2e/pages/utilisation-reports/fee-record-correction/checkFeeRecordCorrectionRequestPage.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (sw-mattw) [reply e2e-tests/tfm/cypress/e2e/pages/utilisation-reports/fee-record-correction/checkFeeRecordCorrectionRequestPage.js]: Sorry my bad, thanks

**Pull request merged**